src/libstd/sync/mod.rs

```diff
@@ -10,10 +10,125 @@
 
 //! Useful synchronization primitives.
 //!
-//! This module contains useful safe and unsafe synchronization primitives.
-//! Most of the primitives in this module do not provide any sort of locking
-//! and/or blocking at all, but rather provide the necessary tools to build
-//! other types of concurrent primitives.
+//! ## The need for synchronization
+//!
+//! On an ideal single-core CPU, the timeline of events happening in a program
+//! is linear, consistent with the order of operations in the code.
+//!
+//! Considering the following code, operating on some global static variables:
+//!
+//! ```rust
+//! # static mut A: u32 = 0;
```

Comment:
Why hiding them? Also: they're not global since they're inside a function. If you don't declare the `main` function, then all code is put inside it (except `extern crate` declarations).

Reply:
> Why hiding them?

I was trying to keep the example code clean and focus on the executed instructions. But I guess they won't hurt readability.

> Also: they're not global since they're inside a function.

Thanks, should be fixed now.